temporal/api/cloud/identity/v1/message.proto

```diff
@@ -66,6 +66,37 @@ message User {
     // The date and time when the user was created
     google.protobuf.Timestamp created_time = 7;
     // The date and time when the user was last modified
-    // Will not be set if the user has never been modified.
+    // Will not be set if the user has never been modified
     google.protobuf.Timestamp last_modified_time = 8;
 }
+
+message UserGroupSpec {
+    // The name of the group as defined in the customer's IdP (e.g. Google group name in Google Workspace)
+    // The name is immutable. Once set, it cannot be changed
+    string name = 1;
```

Comment:
i wonder if we need a name and an "internal id"

Reply:
yes there's an internal id, it's not on the spec object though because it's set by our system (just like user). you can find that id on the `UserGroup` struct